architecture/modules/frontend_bucket/main.tf

```diff
@@ -67,6 +67,38 @@ resource "google_compute_global_forwarding_rule" "frontend_fwd_rule_443" {
   ip_address  = google_compute_global_address.lb_ip.address
 }
 
+// HTTP redirect to HTTPS
+// From https://stackoverflow.com/a/66256756
+
+// URL Map that simply changes the redirected request from HTTP to HTTPS
+resource "google_compute_url_map" "http-redirect" {
```

Comment:
Prefer underscores to hyphens separating words in resource names.

Reply:
Agreed. Same goes for the other resources here.

Reply:
Okay, quick question tho - is this convention for names that are part of the resource block itself or the names that are an attribute of the block? That probably sounds a little vague - I am not sure what each of the two names are called.

Reply:
Technically the bit up top is defining the "resource address" the last part of which is referred to by Hashicorp as the name. The resources (or datablocks themselves) can also have a name parameter, which adds to the confusion. In short:
- Resource name (part of address) needs to have underscores (by convention, if not enforces).
- Parameter name can really be what you like, but if there's no good reason to use hyphens then default to underscores anyway. TF/GCP will usually complain if you try and use invalid characters in fields.